core/src/rpc.rs

```diff
@@ -112,36 +112,30 @@ impl JsonRpcRequestProcessor {
             Some(config) => config.commitment,
         };
 
-        let slot = match commitment_level {
+        let mut slot = self
+            .block_commitment_cache
+            .read()
+            .unwrap()
+            .slot_with_commitment(commitment_level);
+
+        match commitment_level {
             CommitmentLevel::Recent => {
-                let slot = r_bank_forks.highest_slot();
+                if let Some(descendant_slot) = r_bank_forks.highest_descendant(slot) {
+                    slot = descendant_slot;
+                }
                 debug!("RPC using working_bank: {:?}", slot);
```

Comment:
Do you mind updating `working_bank`? As long as that method exists in BankForks, it has a specific meaning.